File: omnes.Web/Modules/ServerTypes/Parametros/PaisesRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface PaisesRow {
    IdPais?: number;
    NombrePais?: string;
    Baja?: boolean;
}

export abstract class PaisesRow {
    static readonly idProperty = 'IdPais';
    static readonly nameProperty = 'NombrePais';
    static readonly localTextPrefix = 'Parametros.Paises';
    static readonly deletePermission = 'SuperAdmin';
    static readonly insertPermission = 'SuperAdmin';
    static readonly readPermission = 'SuperAdmin';
    static readonly updatePermission = 'SuperAdmin';

    static readonly Fields = fieldsProxy<PaisesRow>();
}

File: omnes.Web/Modules/Parametros/Paises/PaisesRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace omnes.Parametros;

[ConnectionKey("Default"), Module("Parametros"), TableName("Paises")]
[DisplayName("Paises"), InstanceName("Paises")]
[ReadPermission("SuperAdmin")]
[ModifyPermission("SuperAdmin")]
public sealed partial class PaisesRow : Row<PaisesRow.RowFields>, IIdRow, INameRow
{
    class RowTemplate
    {
        [DisplayName("Id Pais"), Identity, IdProperty]
        public int? IdPais { get; set; }

        [DisplayName("Nombre Pais"), Size(200), NotNull, QuickSearch, NameProperty]
        public string NombrePais { get; set; }

        [DisplayName("Baja"), NotNull]
        public bool? Baja { get; set; }
    }
}